src/nvim/api/window.c

```diff
@@ -472,6 +472,48 @@ void nvim_win_config(Window window, Integer width, Integer height,
   }
 }
 
+/// Return window configuration of a window opened via |nvim_open_win()|.
+///
+/// Return a dictionary containing the same options that can be given to
+/// |nvim_open_win()| plus the height and the width of the window.
+///
+/// @param window Window handle
+/// @param[out] err Error details, if any
+/// @return Window configuration
+Dictionary nvim_win_get_config(Window window, Error *err)
+  FUNC_API_SINCE(6)
+{
+  Dictionary rv = ARRAY_DICT_INIT;
+
+  win_T *wp = find_window_by_handle(window, err);
+  if (!wp) {
+    return rv;
+  }
+
+  PUT(rv, "height", INTEGER_OBJ(wp->w_height));
```

Comment:
Maybe we should change `nvim_win_set_config` to take optional width/height keys in the dict at the same time, instead of separate parameters, it would the simplest way to support this.